app/src/main/java/com/example/touchcar/presentation/main_menu/recycler/MainMenuAdapter.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.touchcar.presentation.main_menu.recycler
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.example.touchcar.R
+import com.example.touchcar.domain.entity.Manufacturer
+import java.util.*
+import java.util.zip.Inflater
+
+class MainMenuAdapter() : RecyclerView.Adapter<MainMenuViewHolder>() {
+
+    var manufacturers: List<Manufacturer> = Collections.emptyList();
+
+
+    fun updateViewHolder(manufacturers: List<Manufacturer>) {
+        val callback = MainMenuDiffCallback(this.manufacturers, manufacturers)
+        val diffResult: DiffUtil.DiffResult = DiffUtil.calculateDiff(callback)
+        this.manufacturers = manufacturers
+        diffResult.dispatchUpdatesTo(this)
+    }
```

Comment:
Давай упростим задачу: на каждое изменение поля будет запускаться эта логика. Переопределим сеттер у поля со списком:
```suggestion
    var manufacturers: List<Manufacturer> = Collections.emptyList();
        set(value) {
            val callback = MainMenuDiffCallback(field, value)
            val diffResult: DiffUtil.DiffResult = DiffUtil.calculateDiff(callback)
            field = value
            diffResult.dispatchUpdatesTo(this)
        }
```